### ISSUE TITLE
Another RSFetcher Failure

### ISSUE THREAD

**Issue by adamkrellenstein** [CounterpartyXCP/counterparty-core#2280]
got the same RSFetcher error but this time during the parsing (the parser is not stopped and restarted but just fails to fetch the next block):
```
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.469+00:00 - [  DEBUG] - Looking for Block 863453 in prefetch queue...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.469+00:00 - [  DEBUG] - Block 863453 retrieved from queue. (Queue: 0/20)
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.508+00:00 - [  DEBUG] - Block 863453 fetched. (0.039929s)
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.512+00:00 - [  INFO] - Block 863453
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.516+00:00 - [  EVENT] - Block 863453 - NEW_BLOCK [block_index=863453 block_hash=00000000000000000002296a26eb53adf99941124e12cd0ebf36fc16cb2aca68 block_time=1727668723 previous_block_hash=00000000000000000002e8aa5aaeca2b3a5687dca33ae3a1c389231f9f989d05 difficulty=386084628]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.622+00:00 - [  EVENT] - Block 863453 - NEW_TRANSACTION [tx_index=2755655 tx_hash=7ab80f4 block_index=863453 block_hash=00000000000000000002296a26eb53adf99941124e12cd0ebf36fc16cb2aca68 block_time=1727668723 source=1Hg93NK7 destination= btc_amount=0 fee=1000 data=0200005f4761ed29a6000000000000000100c5734e0463debda06448ee6d812a17cadb71f76a utxos_info=]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.647+00:00 - [  EVENT] - Block 863453 - DEBIT [block_index=863453 address=1Hg93NK7 utxo=None utxo_address=None asset=THREEPEPES quantity=1 action=send event=7ab80f4d8b496069480b2ce69239df25c5895a84abf4719e2cf8f625ae4403c7 tx_index=2755655]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.654+00:00 - [  EVENT] - Block 863453 - CREDIT [block_index=863453 address=1K12FJiK utxo=None utxo_address=None asset=THREEPEPES quantity=1 calling_function=send event=7ab80f4d8b496069480b2ce69239df25c5895a84abf4719e2cf8f625ae4403c7 tx_index=2755655]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.665+00:00 - [  EVENT] - Block 863453 - ENHANCED_SEND [tx_index=2755655 tx_hash=7ab80f4 block_index=863453 source=1Hg93NK7 destination=1K12FJiK asset=THREEPEPES quantity=1 status=valid memo=None]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.669+00:00 - [  INFO] - Block 863453 - Send (Enhanced) THREEPEPES from 1Hg93NK7 to 1K12FJiK (7ab80f4) [valid]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.677+00:00 - [  EVENT] - Block 863453 - TRANSACTION_PARSED [tx_index=2755655 tx_hash=7ab80f4 supported=True]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.682+00:00 - [  EVENT] - Block 863453 - BLOCK_PARSED [block_index=863453 ledger_hash=a992857 txlist_hash=e67a37d messages_hash=7f2e0b6 transaction_count=1]
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.683+00:00 - [  INFO] - Block 863453 - Parsing complete. L: a992857, TX: e67a37d, M: 7f2e0b6 (0.17s)
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.688+00:00 - [  DEBUG] - Block 863453/863620 parsed, for 447 blocks in 0h 6m 10s.
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.689+00:00 - [  DEBUG] - Looking for Block 863454 in prefetch queue...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.692+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.692+00:00 - [  EVENT] - API Watcher - Event parsed: 17738864 NEW_BLOCK
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.794+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.830+00:00 - [  DEBUG] - API Watcher - 53 mempool events synchronized
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.837+00:00 - [  EVENT] - API Watcher - Event parsed: 17738865 NEW_TRANSACTION
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.850+00:00 - [  EVENT] - API Watcher - Event parsed: 17738867 CREDIT
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.855+00:00 - [  EVENT] - API Watcher - Event parsed: 17738868 DISPENSER_UPDATE
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.864+00:00 - [  EVENT] - API Watcher - Event parsed: 17738869 DISPENSE
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.872+00:00 - [  EVENT] - API Watcher - Event parsed: 17738870 TRANSACTION_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.877+00:00 - [  EVENT] - API Watcher - Event parsed: 17738871 BLOCK_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.895+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.997+00:00 - [  DEBUG] - API Watcher - 6 mempool events synchronized
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:47.999+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.101+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.124+00:00 - [  EVENT] - API Watcher - Event parsed: 17738872 NEW_BLOCK
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.128+00:00 - [  EVENT] - API Watcher - Event parsed: 17738873 BLOCK_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.202+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.305+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.343+00:00 - [  EVENT] - API Watcher - Event parsed: 17738874 NEW_BLOCK
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.346+00:00 - [  EVENT] - API Watcher - Event parsed: 17738875 BLOCK_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.406+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:06:48.508+00:00 - [  DEBUG] - Block 863454 not found in prefetch queue. Waiting...
```

block 863453 is fetched normally and block 863454 is not found
5:24
the server is since stuck on "Block 863454 not found in prefetch queue. Waiting..."

**Comment by adamkrellenstein** [comment]
```
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.311+00:00 - [  INFO] - Block 863477 - Parsing complete. L: 73fd065, TX: fa16063, M: 177a0e7 (0.67s)
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.315+00:00 - [  DEBUG] - Block 863477/863624 parsed, for 24 blocks in 0h 2m 2s.
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.316+00:00 - [  DEBUG] - Looking for Block 863478 in prefetch queue...
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.316+00:00 - [  DEBUG] - Block 863478 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.379+00:00 - [  EVENT] - API Watcher - Event parsed: 17739094 NEW_BLOCK
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.395+00:00 - [  EVENT] - API Watcher - Event parsed: 17739095 NEW_TRANSACTION
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.406+00:00 - [  EVENT] - API Watcher - Event parsed: 17739096 NEW_TRANSACTION
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.416+00:00 - [  EVENT] - API Watcher - Event parsed: 17739097 NEW_TRANSACTION
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.417+00:00 - [  DEBUG] - Block 863478 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.440+00:00 - [  EVENT] - API Watcher - Event parsed: 17739098 DISPENSER_UPDATE
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.447+00:00 - [  EVENT] - API Watcher - Event parsed: 17739099 TRANSACTION_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.460+00:00 - [  EVENT] - API Watcher - Event parsed: 17739100 DISPENSER_UPDATE
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.466+00:00 - [  EVENT] - API Watcher - Event parsed: 17739101 TRANSACTION_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.477+00:00 - [  EVENT] - API Watcher - Event parsed: 17739102 DISPENSER_UPDATE
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.482+00:00 - [  EVENT] - API Watcher - Event parsed: 17739103 TRANSACTION_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.485+00:00 - [  EVENT] - API Watcher - Event parsed: 17739104 BLOCK_PARSED
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.518+00:00 - [  DEBUG] - Block 863478 not found in prefetch queue. Waiting...
counterparty-core-counterparty-core-1 | 2024-10-01T09:27:59.603+00:00 - [  ERROR] - Error prefetching block: Stopped error
```

`Error prefetching block: Stopped error`: looks like the fetcher stopped for an unknown reason

**Comment by adamkrellenstein** [comment]
Hopefully addressed by hopefully fixed by #2285
